### DISCUSSION TITLE
Wild cards (*) for fixtures

### DISCUSSION THREAD

**Discussion by GregoireGaonach** [saloonphp/saloon#394] [Ideas]
Hello everyone!

SaloonPHP has been absolutely amazing for my project, so thank you from the bottom of my heart.

Just an idea I wanted to share. I am writing some tests for the European Parliament API endpoint. To get all the data, there is a need to do nested calls, which I can easily handle with nested actions.

Anyway, on the tests, because of this, I need to design a lot of fixtures to have the test pass as close to reality as possible. At the moment it looks like this:

```php
Saloon::fake([
    'data.europarl.europa.eu/api/v1/corporate-bodies/ep-9' => MockResponse::fixture('corporate-body-ep-9'),
    'data.europarl.europa.eu/api/v1/corporate-bodies/6417' => MockResponse::fixture('corporate-body-6417'),
    'data.europarl.europa.eu/api/v1/corporate-bodies/5116' => MockResponse::fixture('corporate-body-5116'),
    'data.europarl.europa.eu/api/v1/corporate-bodies/6359' => MockResponse::fixture('corporate-body-6359'),
    ...
    ...
)];
```

What I would love to be able to do is something like this:

```php
Saloon::fake([
    'data.europarl.europa.eu/api/v1/corporate-bodies/*' => MockResponse::fixture('corporate-body-*'),
)];
```

As far as I know this is not possible at the moment, wildcards `(*)` are only supported for the endpoint and not for the fixture. 

I would be happy to try to implement this feature if you think it is a good idea, but I would have to dive into the codebase first.

Thanks again for the great work!

**Comment by MCKLtech** [comment]
I have (I think), the very same problem today. My ideal test would look like this:

```
Saloon::fake([
            'users/112' => MockResponse::make([], 404),
            'users?query[email]=bob@example.com*' => MockResponse::make([...])
]);
```

But when I run the test, the URL which is called is:

```
https://api.example.com/api/users?query%5Bemail%5D=bob%40example.com&limit=1&page=1&per_page=100
```

which results in:

```
Saloon was unable to guess a mock response for your request {URL} consider using a wildcard url mock or a connector mock.

```

Basically, I don't care about the end of the URL after the email parameter as they won't change the return I'm expecting.

To add, the test still fails, even if I use the full URL (URL encoded or not) e.g.

```
Saloon::fake([
            'users/112' => MockResponse::make([], 404),
            'users?query%5Bemail%5D=bob%40example.com&limit=1&page=1&per_page=100' => MockResponse::make([...])
]);
```
